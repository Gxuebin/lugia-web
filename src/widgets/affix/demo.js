/**
 *
 * create by guorg
 *
 * @flow
 */
import * as React from 'react';
import styled from 'styled-components';
import Affix from './affix';
import Button from '../button';

export default class AffixDemo extends React.Component {
  constructor() {
    super();
    this.EleRef = null;
    this.Ele2Ref = null;
    this.state = {
      display: true,
    };
  }
  render() {
    const { display } = this.state;
    return (
      <div>
        {display ? (
          <div style={{ width: '200px', height: '1200px' }}>
            <div style={{ width: '200px', height: '200px' }} />
            <Affix offsetTop={50}>
              <Button>affix-top</Button>
            </Affix>

            <div
              style={{ width: '200px', height: '200px', overflowY: 'scroll' }}
              ref={node => (this.EleRef = node)}
            >
              <div style={{ height: '400px' }}>
                <div style={{ width: '20px', height: '100px' }} />
                <Affix offsetTop={50} target={() => this.EleRef}>
                  <Button>affix-top</Button>
                </Affix>
              </div>
            </div>
            <div
              style={{ width: '200px', height: '200px', overflowY: 'scroll' }}
              ref={node => (this.Ele2Ref = node)}
            >
              <div style={{ height: '400px' }}>
                <div style={{ width: '20px', height: '200px' }} />
                <Affix offsetBottom={50} target={() => this.Ele2Ref}>
                  <Button>affix2-bottom</Button>
                </Affix>
              </div>
            </div>

            <div style={{ width: '200px', height: '500px' }} />
            <Affix offsetBottom={50}>
              <Button>affix-bottom</Button>
            </Affix>
            <div style={{ width: '200px', height: '500px' }} />
          </div>
        ) : (
          <div style={{ width: '200px', height: '1200px' }} />
        )}
      </div>
    );
  }
}
