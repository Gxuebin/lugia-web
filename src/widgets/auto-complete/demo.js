import * as React from 'react';
import AutoComplete from './';

const data = [
  'Nikcy Romero',
  'Armin van Buuren',
  'Hardwell',
  'Zedd',
  'Kazze',

  'Vicetone',
  'Martin Garrix',
  'David Guetta',
  'The Chainsmokers',
  'Kygo',

  'Axwell ^ Ingrosso',
  'Dimitri Vegas & Like Mike',
  'Calvin Harris',
  'Avicci',
  'Fedde Le Grand',

  'Tiesto',
  'Snoop Dogg',
  'Bassjackers',
  'Sebastian Ingrosso',
  'Swedish House Mafia',
  'Alesso',
  'Afrojack',
  'Knife Party',
  'Dannic',
  'R3hab',
];

export default class AutoCompleteBounded extends React.Component<any, any> {
  constructor(props) {
    super(props);
    this.state = {
      menuData: data,
    };
  }
  static getDerivedStateFromProps(props: any, state: any) {
    const hasValueInProps = 'value' in props;
    const value = hasValueInProps ? props.value : state ? state.value : props.defaultValue;
    if (!state) {
      return {
        value,
      };
    }
    return {
      value,
    };
  }

  render() {
    const { menuData, value } = this.state;
    return <AutoComplete showOldValue value={value} data={menuData} onChange={this.onChange} />;
  }

  onChange = (value: string) => {
    this.search(value);
    this.setState({ value });
  };

  search(query: string) {
    let menuData;
    let rowSet = [];
    const len = data.length;

    for (let i = 0; i < len; i++) {
      const row = data[i];
      if (this.searchValue(query, row)) {
        rowSet.push(row);
      }
      if (query === row) {
        rowSet = [];
        break;
      }
    }

    if (rowSet.length === len) {
      menuData = data;
    } else {
      menuData = rowSet;
    }
    this.setState({ menuData });
  }

  searchValue = (query: string, row: string): boolean => {
    return row.indexOf(query) !== -1;
  };
}

// export default class Demo extends React.Component<any, any> {
//   constructor(props) {
//     super(props);
//     this.state = {
//       menuData: data,
//       value: '',
//     };
//   }
//   render() {
//     return (
//       <AutoComplete
//         data={this.state.menuData}
//         onChange={this.onChange}
//         placeholder={'请输入'}
//         showOldValue={true}
//       />
//     );
//   }

//   onChange = value => {
//     this.search(value);
//   };

//   search(query) {
//     let menuData;
//     let rowSet = [];
//     const len = data.length;

//     for (let i = 0; i < len; i++) {
//       const row = data[i];
//       if (query === row) {
//         rowSet = [];
//         break;
//       }

//       if (this.searchValue(query, row)) {
//         rowSet.push(row);
//       }
//     }

//     if (rowSet.length === len) {
//       menuData = data;
//     } else {
//       menuData = rowSet;
//     }
//     this.setState({ menuData });
//   }

//   searchValue = (query, row) => {
//     return row.indexOf(query) !== -1 || row === query;
//   };
// }
