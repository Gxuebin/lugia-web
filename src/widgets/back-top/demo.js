/**
 *
 * create by guorg
 *
 * @flow
 */
import * as React from 'react';
import styled from 'styled-components';
import BackTop from './back-top';
import Widget from '../consts/index';
import Theme from '../theme';

const Demo = styled.div`
  & > span > div > div {
    bottom: ${props => 50 + props.cur * 50}px;
  }
`;
const DemoBack = styled.div`
  width: 40px;
  height: 40px;
  line-height: 40px;
  text-align: center;
  border-radius: 4px;
  border: 1px solid #e8e8e8;
`;
export default class AffixDemo extends React.Component<any, any> {
  EleRef: any;
  render() {
    const view = {
      [Widget.BackTop]: {
        normal: {
          color: 'red',
          background: { backgroundColor: 'orange' },
          width: 100,
          height: 100,
          borderRadius: 100,
          opacity: 0.7,
          border: {
            borderWidth: 1,
            borderStyle: 'solid',
            borderColor: 'red',
          },
        },
      },
    };
    return (
      <div style={{ height: '1600px' }}>
        <BackTop />
        <Demo cur={1}>
          <BackTop>
            <DemoBack>UP</DemoBack>
          </BackTop>
        </Demo>
        <Demo cur={2}>
          <Theme config={view}>
            <BackTop />
          </Theme>
        </Demo>
        <div
          style={{ width: '200px', height: '200px', overflowY: 'scroll', marginTop: '400px' }}
          ref={node => (this.EleRef = node)}
        >
          <Demo cur={3}>
            <div style={{ width: '20px', height: '300px' }} />
            <BackTop visibilityHeight={80} target={() => this.EleRef} />
          </Demo>
        </div>
      </div>
    );
  }
}
