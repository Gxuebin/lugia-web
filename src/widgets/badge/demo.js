/**
 *
 * create by liangguodong on 2018/8/27
 *
 * @flow
 */
import * as React from 'react';
import styled from 'styled-components';
import Badge from './';
import { Col, Row } from '../grid';
import Widget from '../consts';
import Theme from '../theme';
import Icon from '../icon';

const Box = styled.div`
  width: 40px;
  height: 40px;
  background: #ccc;
  margin-left: 10px;
`;
const Wrapper = styled.div`
  margin-left: 10px;
  margin-top: 50px;
`;

class Turn extends React.Component<any, any> {
  constructor(props) {
    super(props);
    this.state = {
      count: 0,
    };
  }

  click = type => () => {
    const count = type === 'plus' ? this.state.count + 1 : this.state.count - 1;
    this.setState({ count });
  };

  render() {
    const config = {
      [Widget.Badge]: {
        children: {
          [Widget.NumberTurn]: {
            normal: { position: { top: -5, right: -5 } },
          },
        },
      },
    };
    return (
      <Wrapper>
        <Theme config={config}>
          <Row>
            <Col span={4}>
              <Badge count={this.state.count} showZero overflowCount={99}>
                <Box />
              </Badge>
              <Icon
                style={{ fontSize: '2em' }}
                iconClass="lugia-icon-reminder_plus_square_o"
                onClick={this.click('plus')}
              />
              <Icon
                style={{ fontSize: '2em' }}
                iconClass="lugia-icon-reminder_minus_square_o"
                onClick={this.click('minus')}
              />
            </Col>
          </Row>
        </Theme>
      </Wrapper>
    );
  }
}

export default () => {
  const view = {
    [Widget.Badge]: {
      normal: {
        position: { top: -5, right: 0 },
        background: { backgroundColor: 'orange' },
        color: 'black',
      },
    },
  };
  return (
    <div>
      <Theme
        config={{
          [Widget.Badge]: {
            children: {
              [Widget.NumberTurn]: {
                normal: { position: { top: -5, right: -5 } },
              },
            },
          },
        }}
      >
        <Wrapper>
          <Row>
            <Col span={4}>
              <Badge count={4}>
                <Box />
              </Badge>
            </Col>
            <Col span={4}>
              <Badge count={99}>
                <Box />
              </Badge>
            </Col>
            <Col span={4}>
              <Badge>
                <Box />
              </Badge>
            </Col>
          </Row>
        </Wrapper>
        <Wrapper>
          <Theme config={view}>
            <Row>
              <Col span={4}>
                <Badge count={0}>
                  <Box />
                </Badge>
              </Col>
              <Col span={4}>
                <Badge showZero>
                  <Box />
                </Badge>
              </Col>
              <Col span={4}>
                <Badge showZero count={0}>
                  <Box />
                </Badge>
              </Col>
              <Col span={4}>
                <a href="www.baidu.com">
                  <Badge>
                    <Box />
                  </Badge>
                </a>
              </Col>
            </Row>
          </Theme>
        </Wrapper>
        <Wrapper>
          <Row>
            <Col span={4}>
              <Theme
                config={{
                  green: {
                    normal: {
                      position: { top: -5, right: 2 },
                      background: { backgroundColor: 'green' },
                    },
                  },
                }}
              >
                <Badge viewClass="green">
                  <Box />
                </Badge>
              </Theme>
            </Col>
            <Col span={4}>
              <Theme
                config={{
                  purple: {
                    normal: {
                      position: { top: -5, right: 2 },
                      background: { backgroundColor: 'purple' },
                    },
                  },
                }}
              >
                <Badge viewClass="purple">
                  <Box />
                </Badge>
              </Theme>
            </Col>
            <Col span={4}>
              <Theme
                config={{
                  yellow: {
                    normal: {
                      position: { top: -5, right: 2 },
                      background: { backgroundColor: 'yellow' },
                    },
                  },
                }}
              >
                <Badge viewClass="yellow">
                  <Box />
                </Badge>
              </Theme>
            </Col>
            <Col span={4}>
              <Theme
                config={{
                  blue: {
                    normal: {
                      position: { top: -5, right: 2 },
                      background: { backgroundColor: 'blue' },
                    },
                  },
                }}
              >
                <Badge viewClass="blue">
                  <Box />
                </Badge>
              </Theme>
            </Col>
          </Row>
        </Wrapper>
        <Turn />
      </Theme>
    </div>
  );
};
