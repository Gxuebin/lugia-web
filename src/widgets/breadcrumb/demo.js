import * as React from 'react';
import Breadcrumb from './index';
import styled from 'styled-components';
import Icon from '../icon';
import { HashRouter as Router, Route, Switch, Link, withRouter } from 'react-router-dom';

const Hr = styled.div`
  height: 2px;
  background: orange;
  margin: 20px 0;
`;

const CommonIcon = styled(Icon)`
  color: #000;
  border-radius: 50%;
`;

const routes = [
  {
    path: 'index',
    breadcrumbName: '首页',
  },
  {
    path: 'first',
    breadcrumbName: '一级面包屑',
  },
  {
    path: 'second',
    breadcrumbName: '二级面包屑',
  },
  {
    path: 'third',
    breadcrumbName: '当前页面',
  },
];

export default class Demo extends React.Component<any, any> {
  render() {
    return (
      <div>
        {/*Breadcrumb 和 Breadcrumb.Item配合使用，使用href属性时，可跳转 */}
        <Breadcrumb separator={'>'}>
          <Breadcrumb.Item href="">主页</Breadcrumb.Item>
          <Breadcrumb.Item href="">一级菜单</Breadcrumb.Item>
          <Breadcrumb.Item href="">二级菜单</Breadcrumb.Item>
          <Breadcrumb.Item>三级菜单</Breadcrumb.Item>
        </Breadcrumb>
        <Hr />
        {/* 传入routes路由属性，使用默认的 renderItem函数,生成面包屑组件 */}
        <Breadcrumb separator={'>'} routes={routes} />
        <Hr />
        {/* 传入Icon图标 */}
        <Breadcrumb>
          <Breadcrumb.Item href="">
            <CommonIcon iconClass="lugia-icon-logo_chrome" type="home" />
            <span>一级面包屑菜单 </span>
          </Breadcrumb.Item>
          <Breadcrumb.Item href="">
            <span>二级面包屑菜单</span>
            <CommonIcon iconClass="lugia-icon-financial_smile" type="user" />
          </Breadcrumb.Item>
          <Breadcrumb.Item>
            <span>二级面包屑菜单</span>
            <CommonIcon iconClass="lugia-icon-logo_apple" type="user" />
          </Breadcrumb.Item>
        </Breadcrumb>
      </div>
    );
  }
}

// //       demo4
// // !!!!!!!!!!!!!!!!!!!这个组件这么使用有bug，不懂什么原因
// // 使用的自定义设置的 renderItem回调函数,传入routes路由属性，生成面包屑组件=====================
// import * as React from 'react';
// import Breadcrumb from './index';
// import { Link, Router } from 'react-router-dom';

// const routes = [
//   {
//     path: 'index',
//     breadcrumbName: '首页',
//   },
//   {
//     path: 'first',
//     breadcrumbName: '一级面包屑',
//   },
//   {
//     path: 'second',
//     breadcrumbName: '二级面包屑',
//   },
//   {
//     path: 'third',
//     breadcrumbName: '当前页面',
//   },
// ];

// function renderItem(route, params, routes, paths) {
//   const last = routes.indexOf(route) === routes.length - 1;
//   return last ? (
//     <span>{route.breadcrumbName}</span>
//   ) : (
//     <Link to={paths.join('/')}>{route.breadcrumbName}</Link>
//   );
// }
// export default class Demo extends React.Component<any, any> {
//   render() {
//     return (
//       <Router>
//         <Breadcrumb renderItem={renderItem} routes={routes} />
//       </Router>
//     );
//   }
// }

//     demo6
//  给路由传入params 参数
// import * as React from 'react';
// import Breadcrumb from './index';
// const routes = [
//   {
//     name: 'home',
//     breadcrumbName: 'Home',
//     path: '/',
//   },
//   {
//     name: 'apps',
//     breadcrumbName: 'Application List',
//     path: 'apps',
//   },
//   {
//     name: 'app',
//     breadcrumbName: 'Application:id',
//     path: ':id',
//   },
//   {
//     name: 'detail',
//     breadcrumbName: 'Detail',
//     path: 'detail',
//   },
// ];

// export default class Demo extends React.Component<any, any> {
//   render() {
//     return <Breadcrumb params={{ id: 1, class: 2 }} routes={routes} />;
//   }
// }
