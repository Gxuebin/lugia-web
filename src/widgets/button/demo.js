/**
 *
 * create by ligx
 *
 * @flow
 */
import React from 'react';
import Button from './index';

export default () => {
  return <Button type="danger">hello</Button>;
};
