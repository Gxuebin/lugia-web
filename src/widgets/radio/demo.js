/**
 *
 * create by guorg
 *
 * @flow
 */
import React from 'react';
import styled from 'styled-components';
import { getBorder } from '@lugia/theme-css-hoc';
import Theme from '../theme';
import Widget from '../consts';
import Radio from './';

const RadioGroup = Radio.Group;
const RadioButton = Radio.Button;

const Wrapper = styled.div`
  margin-left: 50px;
  width: 50%;
`;
const data = [
  { text: 'radio1', value: 'Apple' },
  { text: 'radio2', value: 'Pear' },
  { text: 'radio3', value: 'Orange', disabled: true },
];
const view = {
  [Widget.RadioGroup]: {
    width: 100,
  },
  register: {
    width: 120,
    color: 'red',
    margin: 20,
  },
};
const radioView = {
  [Widget.RadioGroup]: {
    width: 100,
    color: 'red',
  },
};
const onChange = obj => {
  console.info('obj-demo', obj);
};
export class RadioDemo extends React.Component<any, any> {
  render() {
    const view = {
      [Widget.Radio]: {
        RadioWrap: {
          normal: {
            color: 'red',
            fontSize: 16,
            font: { fontWeight: 200, fontSize: 16 },
            opacity: 0.6,
            width: 100,
            height: 50,
            margin: {
              top: 10,
              bottom: 10,
              left: 10,
              right: 10,
            },
            padding: {
              top: 10,
              bottom: 10,
              left: 10,
              right: 10,
            },
          },
          hover: {
            opacity: 0.4,
          },
        },
        RadioEdge: {
          normal: {
            background: { backgroundColor: 'orange' },
            border: getBorder({ color: 'red', width: 2, style: 'solid' }, { radius: 100 }),
            width: 20,
            height: 20,
          },
          hover: {
            background: { backgroundColor: 'green' },
            border: getBorder({ color: 'orange', width: 2, style: 'solid' }, { radius: 100 }),
          },
          active: {
            background: { backgroundColor: 'yellow' },
            border: getBorder({ color: 'pink', width: 2, style: 'solid' }, { radius: 100 }),
            width: 26,
            height: 26,
          },
        },
        RadioChecked: {
          active: {
            background: { backgroundColor: 'red' },
            width: 18,
            height: 18,
          },
          disabled: {
            background: { backgroundColor: 'red' },
            width: 15,
            height: 15,
          },
          cancel: {
            background: { backgroundColor: 'green' },
            width: 12,
            height: 12,
          },
        },
      },
    };
    const newView = {
      [Widget.Radio]: {
        RadioWrap: {
          hover: {
            opacity: 0.4,
          },
        },
        RadioEdge: {
          disabled: {
            background: { backgroundColor: 'orange' },
            border: getBorder({ color: 'red', width: 2, style: 'solid' }, { radius: 100 }),
          },
        },
        RadioChecked: {
          disabled: {
            background: { backgroundColor: 'red' },
            width: 8,
            height: 8,
          },
          cancel: {
            background: { backgroundColor: 'green' },
            width: 12,
            height: 12,
          },
        },
      },
    };
    return (
      <div>
        <Wrapper>
          <p>theme normal margin padding</p>
          <Theme config={view}>
            <Radio value="1">Radio</Radio>
          </Theme>
        </Wrapper>
        <Wrapper>
          <p>theme checked cancel</p>
          <Theme config={view}>
            <Radio value="1" checked cancel>
              Radio
            </Radio>
          </Theme>
        </Wrapper>
        <Wrapper>
          <p>theme checked disabled</p>
          <Theme config={newView}>
            <Radio value="1" checked disabled>
              Radio
            </Radio>
          </Theme>
        </Wrapper>
        <Wrapper>
          <p>theme disabled</p>
          <Theme config={newView}>
            <Radio value="1" disabled>
              Radio
            </Radio>
          </Theme>
        </Wrapper>
        <Wrapper>
          <p>theme cancel</p>
          <Theme config={newView}>
            <Radio value="1" checked cancel>
              Radio
            </Radio>
          </Theme>
        </Wrapper>
        <Wrapper>
          <p>checked</p>
          <Radio checked>Radio</Radio>
        </Wrapper>
        <Wrapper>
          <p>defaultChecked</p>
          <Radio defaultChecked>Radio</Radio>
        </Wrapper>
        <Wrapper>
          <p>normal</p>
          <Radio>Radio</Radio>
        </Wrapper>
        <Wrapper>
          <p>disabled</p>
          <Radio value="1" disabled>
            Radio
          </Radio>
          <Radio value="1" checked disabled>
            Radio
          </Radio>
        </Wrapper>
        <Wrapper>
          <p>cancel</p>
          <Radio defaultChecked cancel>
            Radio
          </Radio>
        </Wrapper>
        <Wrapper>
          <p>style default</p>
          <Radio value="1">Radio</Radio>
          <Radio value="2">Radio</Radio>
          <p>style vertical</p>
          <Radio value="1" checked styles="vertical">
            Radio
          </Radio>
          <Radio value="1" styles="vertical">
            Radio
          </Radio>
        </Wrapper>
      </div>
    );
  }
}
export class RadioGroupDemo extends React.Component<any, any> {
  constructor() {
    super();
    this.state = {
      value: 'apple',
      displayValue: 'hello',
    };
  }
  handleChange = (obj: Object) => {
    this.setState({
      value: obj.newValue,
      displayValue: obj.newDisplayValue,
    });
  };
  render() {
    return (
      <div>
        <Wrapper>
          <p>RadioGroup 正常</p>
          <RadioGroup defaultValue="1">
            <Radio value="1">Radio</Radio>
            <Radio value="2">Radio2</Radio>
          </RadioGroup>
          <p>RadioGroup</p>
          <RadioGroup value="1">
            <Radio value="1">Radio</Radio>
            <Radio value="2">Radio2</Radio>
          </RadioGroup>
          <p>vertical</p>
          <RadioGroup defaultValue="1">
            <Radio value="1" styles="vertical">
              Radio
            </Radio>
            <Radio value="2" styles="vertical">
              Radio
            </Radio>
          </RadioGroup>
          <p>data</p>
          <RadioGroup displayFiled="text" defaultValue="Apple" data={data} />
          <p>styled vertical</p>
          <RadioGroup
            onChange={onChange}
            styles="vertical"
            displayFiled="text"
            valueField="value"
            defaultValue="Apple"
            data={data}
          />
        </Wrapper>
        <Wrapper>
          <p>themes</p>
          <Theme config={view}>
            <RadioGroup defaultValue="1" onChange={onChange}>
              <Radio viewClass="register" value="1" styles="vertical">
                Radio
              </Radio>

              <Radio viewClass="register" value="2" styles="vertical">
                Radio
              </Radio>
            </RadioGroup>
          </Theme>
          <p>data theme</p>
          <Theme config={view}>
            <RadioGroup defaultValue="Apple" valueField="value" data={data} />
          </Theme>
        </Wrapper>

        <Wrapper>
          <p>data displayValue</p>
          <RadioGroup
            onChange={onChange}
            displayFiled="text"
            valueField="value"
            value="apples"
            styles="vertical"
            displayValue="DisplayValue"
            data={data}
          />
        </Wrapper>
        <Wrapper>
          <p>data displayValue</p>
          <RadioGroup
            onChange={onChange}
            displayFiled="text"
            valueField="value"
            defaultValue="apples"
            styles="vertical"
            displayValue="DisplayValue"
            data={data}
          />
        </Wrapper>

        <Wrapper>
          <RadioGroup childType="button" onChange={onChange} defaultValue="1">
            <RadioButton value="1">CheckBox1</RadioButton>
            <RadioButton value="2">CheckBox2</RadioButton>
            <RadioButton value="3">CheckBox3</RadioButton>
          </RadioGroup>
        </Wrapper>
        <br />
        <Wrapper>
          <Theme config={radioView}>
            <RadioGroup childType="button" onChange={onChange} defaultValue="1">
              <RadioButton value="1">CheckBox1</RadioButton>
              <RadioButton value="2">CheckBox2</RadioButton>
              <RadioButton value="3">CheckBox3</RadioButton>
            </RadioGroup>
          </Theme>
        </Wrapper>
        <br />
        <Wrapper>
          <RadioGroup
            onChange={onChange}
            data={data}
            defaultValue="apple"
            displayValue="check"
            childType="button"
          />
        </Wrapper>
        <br />
        <Wrapper>
          <RadioGroup
            onChange={this.handleChange}
            data={data}
            value={this.state.value}
            displayValue={this.state.displayValue}
            childType="button"
          />
        </Wrapper>
      </div>
    );
  }
}
