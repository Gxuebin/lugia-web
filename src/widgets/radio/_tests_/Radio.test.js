/**
 * Radio 单元测试
 * create by guorg
 * @flow
 */
import React from 'react';
import chai from 'chai';

import 'jest-styled-components';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Radio from '../radio';
import { RadioDemo } from '../demo';
import renderer from 'react-test-renderer';
import { delay } from '@lugia/react-test-utils';

const { expect: exp } = chai;

Enzyme.configure({ adapter: new Adapter() });

describe('Radio', () => {
  const createMount = (config: Object) => {
    return <Radio {...config}>Radio</Radio>;
  };

  it('Radio css', () => {
    const target = <RadioDemo />;
    expect(renderer.create(target).toJSON()).toMatchSnapshot();
  });

  it('Radio checked', () => {
    const target = mount(createMount({ checked: true }));
    const cmpRadio = target
      .children()
      .at(0)
      .instance();
    expect(cmpRadio.props.checked).toBe(true);
  });

  // it('Radio defaultChecked',() => {
  //   const target =mount(<Radio defaultChecked>Radio</Radio>);
  //   const cmpRadio = target
  //     .children()
  //     .at(0)
  //     .instance();
  //   expect(cmpRadio.state.checked).toBe(true);
  // });

  // it('Radio click',() => {
  //   const target =mount(<Radio checked>Radio</Radio>);
  //   const cmpRadio = target
  //     .children()
  //     .at(0)
  //     .instance();
  //   expect(cmpRadio.state.checked).toBe(true);
  // });

  it('Radio onClick: disabled checked defaultChecked', () => {
    let value = 1;
    const handleClick = () => {
      value = 2;
    };
    function onClick(target: any, result: number) {
      target
        .find('label')
        .at(0)
        .simulate('click');
      expect(value).toBe(result);
    }
    const target = mount(
      createMount({ onChange: handleClick, disabled: true, defaultChecked: true })
    );
    onClick(target, 1);
    const newTarget = mount(createMount({ onChange: handleClick, checked: true }));
    onClick(newTarget, 2);
    const defaultCheckTarget = mount(createMount({ onChange: handleClick, defaultChecked: true }));
    onClick(defaultCheckTarget, 2);
  });
});
