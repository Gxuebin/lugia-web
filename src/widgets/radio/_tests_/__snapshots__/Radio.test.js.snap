// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Radio css 1`] = `
.c1 {
  font-size: 1.4rem;
  line-height: 1.5;
  color: #333333;
  padding: 0;
  list-style: none;
  display: inline-block;
  position: relative;
  white-space: nowrap;
  width: none;
  cursor: pointer;
  display: inline-block;
  margin-right: 2.142857142857143em;
}

.c1:hover > span > span {
  border: 1px solid #684fff;
}

.c6 {
  font-size: 1.4rem;
  line-height: 1.5;
  color: #cccccc;
  padding: 0;
  list-style: none;
  display: inline-block;
  position: relative;
  white-space: nowrap;
  width: none;
  cursor: not-allowed;
  display: inline-block;
  margin-right: 2.142857142857143em;
}

.c6:hover > span > span {
  border: 1px solid #e8e8e8;
}

.c9 {
  font-size: 1.4rem;
  line-height: 1.5;
  color: #333333;
  padding: 0;
  list-style: none;
  display: inline-block;
  position: relative;
  white-space: nowrap;
  width: none;
  cursor: pointer;
  display: block;
  margin-bottom: 0.7142857142857143em;
}

.c9:hover > span > span {
  border: 1px solid #684fff;
}

.c10 {
  font-size: 1.4rem;
  line-height: 1.5;
  color: #333333;
  padding: 0;
  list-style: none;
  display: inline-block;
  position: relative;
  white-space: nowrap;
  width: 8.571428571428571em;
  cursor: pointer;
  display: inline-block;
  margin-right: 2.142857142857143em;
  margin: 1.4285714285714286em;
}

.c10:hover > span > span {
  border: 1px solid red;
}

.c2 {
  margin: 0;
  outline: none;
  line-height: 1;
  vertical-align: text-bottom;
  display: inline-block;
}

.c4 {
  padding-left: 0.7142857142857143em;
}

.c3 {
  position: relative;
  top: 0;
  left: 0;
  display: block;
  width: 1.142857142857143em;
  height: 1.142857142857143em;
  border-radius: 50%;
  border: 1px solid #cccccc;
  background-color: #fff;
  -webkit-transition: all 0.3s;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c5 {
  position: relative;
  top: 0;
  left: 0;
  display: block;
  width: 1.142857142857143em;
  height: 1.142857142857143em;
  border-radius: 50%;
  border: 1px solid #684fff;
  background-color: #fff;
  -webkit-transition: all 0.3s;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c5::after {
  position: absolute;
  width: 0.8571428571428572em;
  height: 0.8571428571428572em;
  left: 0.14285714285714288em;
  top: 0.14285714285714288em;
  border-radius: 0.7142857142857143em;
  display: table;
  border-top: 0;
  border-left: 0;
  content: ' ';
  background-color: #684fff;
}

.c7 {
  position: relative;
  top: 0;
  left: 0;
  display: block;
  width: 1.142857142857143em;
  height: 1.142857142857143em;
  border-radius: 50%;
  border: 1px solid #cccccc;
  background-color: #fff;
  -webkit-transition: all 0.3s;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  background-color: #f2f2f2;
  border: 1px solid #e8e8e8;
  color: #cccccc;
}

.c7::after {
  background-color: none;
}

.c8 {
  position: relative;
  top: 0;
  left: 0;
  display: block;
  width: 1.142857142857143em;
  height: 1.142857142857143em;
  border-radius: 50%;
  border: 1px solid #684fff;
  background-color: #fff;
  -webkit-transition: all 0.3s;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  background-color: #f2f2f2;
  border: 1px solid #e8e8e8;
  color: #cccccc;
}

.c8::after {
  position: absolute;
  width: 0.8571428571428572em;
  height: 0.8571428571428572em;
  left: 0.14285714285714288em;
  top: 0.14285714285714288em;
  border-radius: 0.7142857142857143em;
  display: table;
  border-top: 0;
  border-left: 0;
  content: ' ';
  background-color: #cccccc;
}

.c8::after {
  background-color: #cccccc;
}

.c11 {
  position: relative;
  top: 0;
  left: 0;
  display: block;
  width: 1.142857142857143em;
  height: 1.142857142857143em;
  border-radius: 50%;
  border: 1px solid red;
  background-color: #fff;
  -webkit-transition: all 0.3s;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c11::after {
  position: absolute;
  width: 0.8571428571428572em;
  height: 0.8571428571428572em;
  left: 0.14285714285714288em;
  top: 0.14285714285714288em;
  border-radius: 0.7142857142857143em;
  display: table;
  border-top: 0;
  border-left: 0;
  content: ' ';
  background-color: red;
}

.c0 {
  margin-left: 50px;
  width: 50%;
}

<div>
  <div
    className="c0"
  >
    <p>
      default
    </p>
    <label
      className="c1"
      onClick={[Function]}
    >
      <span
        className="c2"
      >
        <span
          className="c3"
        />
      </span>
      <span
        className="c4"
      >
        Radio
      </span>
    </label>
  </div>
  <div
    className="c0"
  >
    <p>
      checked
    </p>
    <label
      className="c1"
      onClick={[Function]}
    >
      <span
        className="c2"
      >
        <span
          checked={true}
          className="c5"
        />
      </span>
      <span
        className="c4"
      >
        Radio
      </span>
    </label>
  </div>
  <div
    className="c0"
  >
    <p>
      defaultChecked
    </p>
    <label
      className="c1"
      onClick={[Function]}
    >
      <span
        className="c2"
      >
        <span
          checked={true}
          className="c5"
        />
      </span>
      <span
        className="c4"
      >
        Radio
      </span>
    </label>
  </div>
  <div
    className="c0"
  >
    <p>
      disabled
    </p>
    <label
      className="c6"
      disabled={true}
      onClick={[Function]}
    >
      <span
        className="c2"
      >
        <span
          className="c7"
          disabled={true}
        />
      </span>
      <span
        className="c4"
      >
        Radio
      </span>
    </label>
    <label
      className="c6"
      disabled={true}
      onClick={[Function]}
    >
      <span
        className="c2"
      >
        <span
          checked={true}
          className="c8"
          disabled={true}
        />
      </span>
      <span
        className="c4"
      >
        Radio
      </span>
    </label>
  </div>
  <div
    className="c0"
  >
    <p>
      style default
    </p>
    <label
      className="c1"
      onClick={[Function]}
    >
      <span
        className="c2"
      >
        <span
          className="c3"
        />
      </span>
      <span
        className="c4"
      >
        Radio
      </span>
    </label>
    <label
      className="c1"
      onClick={[Function]}
    >
      <span
        className="c2"
      >
        <span
          className="c3"
        />
      </span>
      <span
        className="c4"
      >
        Radio
      </span>
    </label>
    <p>
      style vertical
    </p>
    <label
      className="c9"
      onClick={[Function]}
    >
      <span
        className="c2"
      >
        <span
          checked={true}
          className="c5"
        />
      </span>
      <span
        className="c4"
      >
        Radio
      </span>
    </label>
    <label
      className="c9"
      onClick={[Function]}
    >
      <span
        className="c2"
      >
        <span
          className="c3"
        />
      </span>
      <span
        className="c4"
      >
        Radio
      </span>
    </label>
  </div>
  <label
    className="c10"
    onClick={[Function]}
  >
    <span
      className="c2"
    >
      <span
        checked={true}
        className="c11"
      />
    </span>
    <span
      className="c4"
    >
      Radio
    </span>
  </label>
</div>
`;
